File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cpp
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        vector<int> ans;
        if(root == NULL) return ans;
        int max = INT_MIN;
        queue<TreeNode*> q;
        q.push(root);
        q.push(NULL);
        while(!q.empty()){
            TreeNode* curr = q.front();
            q.pop();
            if(curr == NULL){
                ans.push_back(max);
                max = INT_MIN;
                if(q.empty()) break;
                else q.push(NULL);
            }
            else{
                if(curr -> val > max) max = curr -> val;
                if(curr -> left) q.push(curr -> left);
                if(curr -> right) q.push(curr -> right);
            }
        }
        return ans;
    }
};

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        List<Integer> al = new ArrayList<>();
        bfs(root, al);
        return al;
    }
    void bfs(TreeNode root, List<Integer> al){
        int max = Integer.MIN_VALUE;
        if(root == null)
            return;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        q.add(null);
        while(!q.isEmpty()){
            TreeNode curr = q.remove();
            if(curr == null){
                al.add(max);
                max = Integer.MIN_VALUE;
                if(q.isEmpty())
                    break;
                else
                    q.add(null);
            }
            else{
                if(curr.val > max)
                    max = curr.val;
                if(curr.left != null)
                    q.add(curr.left);
                if(curr.right != null)
                    q.add(curr.right);
            }
        }
    }
}